integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/LogicalRelationVisitor.java

```diff
@@ -91,25 +93,47 @@ public boolean isDefinedAtLogicalPlan(LogicalPlan x) {
 
   private List<D> handleHadoopFsRelation(LogicalRelation x) {
     HadoopFsRelation relation = (HadoopFsRelation) x.relation();
-    return context
-        .getSparkSession()
-        .map(
-            session -> {
-              Configuration hadoopConfig =
-                  session.sessionState().newHadoopConfWithOptions(relation.options());
-              return JavaConversions.asJavaCollection(relation.location().rootPaths()).stream()
-                  .map(p -> PlanUtils.getDirectoryPath(p, hadoopConfig))
-                  .distinct()
-                  .map(
-                      p -> {
-                        // TODO- refactor this to return a single partitioned dataset based on
-                        // static
-                        // static partitions in the relation
-                        return datasetFactory.getDataset(p.toUri(), relation.schema());
-                      })
-                  .collect(Collectors.toList());
-            })
-        .orElse(Collections.emptyList());
+    try {
+      return context
+          .getSparkSession()
+          .map(
+              session -> {
+                Configuration hadoopConfig =
+                    session.sessionState().newHadoopConfWithOptions(relation.options());
+                return JavaConversions.asJavaCollection(relation.location().rootPaths()).stream()
+                    .map(p -> PlanUtils.getDirectoryPath(p, hadoopConfig))
+                    .distinct()
+                    .map(
+                        p -> {
+                          // TODO- refactor this to return a single partitioned dataset based on
+                          // static
+                          // static partitions in the relation
+                          return datasetFactory.getDataset(p.toUri(), relation.schema());
+                        })
+                    .collect(Collectors.toList());
+              })
+          .orElse(Collections.emptyList());
+    } catch (Exception e) {
```

Comment:
@pawel-big-lebowski the specific exception that I'm trying to catch is a private class inside of Databricks.

Would it be worth catching the exception, checking if the exception name matches that specific exception name and then if NOT throwing the exception?

Reply:
Ok. Nów I get the problem. Well, making global OpenLineage class aware of `AzureCredentialNotFoundException` is not super clean, but I cannot see any better approach to handle this (except for creating custom package with custom `LogicalRelationDatasetBuilder`). 

Would love to get @collado-mike opinion on that.

Reply:
@collado-mike Gentle nudge for your opinion on how to handle catching an exception that only shows up on Databricks because they're using a custom class to authenticate against our cloud storage.

Reply:
Sorry. My typical preference for this kind of thing is to try to identify what distinguishes this case from the other case and see if we can pull out a dedicated visitor. Will it always apply to Hadoop FS relations with an ABFS URL? Sometimes the file path points to a mount on the filesystem, right? So we can't always detect ahead of time when this will happen?

Reply:
@collado-mike - Thank you for the reply!  This credential pass through is supposed to be transparent.  It's based on this [standard in Hadoop](https://hadoop.apache.org/docs/stable/hadoop-azure/abfs.html#Custom_OAuth_2.0_Token_Provider).

I don't know if we can tell, at the spark level, that a particular path is going to be affected by this setting. In addition, if you have this configured, it might only be on ONE path you're using. For example, I might have an ABFS path that I'm accessing via a storage key, another ABFS path I'm accessing via OAuth and finally a third that is using this credentials passthrough feature.  All three might be used in the same query.

Does OpenLineage already handle this [credentials passthrough idea on AWS](https://docs.databricks.com/security/credential-passthrough/iam-passthrough.html) by any chance?

Reply:
Thanks for the details here. No, we don't have any special handling for the AWS credentials passthrough either. I don't like the reflection based exception handling, but I think that until we have more general use cases, it's the most straightforward way to handle this case. It would be great to add the AWS use case to the list of issues so we can test and maybe come up with a more general way of addressing this.

Reply:
@collado-mike - Does that mean we can accept this PR for now?  I'll add an issue around "Handling pass through credentials more generically in S3 and ABFS filesystems".

Reply:
Yeah, I haven't looked thoroughly at this, but I think the custom exception handling is fine as is